text/0000-comprehensive-changelogs.md

```diff
@@ -0,0 +1,145 @@
+
+This could potentially be enhanced by having gatsbot post a comment about 'how to title your PR' for PRs that don't match conventional commit format.
+
+Pros:
+
+- Minimal setup time
+
+Cons:
+
+- 'Uncategorised' is not a very useful category
+
+### 3. Don't categorise commits
+
+Drop usage of conventional commits style PR titles, and just list all PRs related to each package version. [See an example from artsy](https://github.com/artsy/emission/blob/master/CHANGELOG.md), along with [discussion of their implementation](https://github.com/artsy/reaction/issues/1095).
+
+Pros:
+
+- Minimal setup time
+- All PRs will get added to the changelog
+
+Cons:
+
+- Potentially less useful than having a categorised changelog
+
+### 4. Add a CI step that fails if a PR title doesn't match conventional commit style
+
+Pros:
+
+- Ensures all PRs are categorised
+- Allows contributors to categorise their own commits
+
+Cons:
+
+- Adds an 'invisible' barrier to getting your PR merged
+- Is there a way to trigger a new CI run after a PR title is updated?
+
+### 5. Use gatsbot / Peril to categorise commits
+
+gatsbot would comment on a PR with a list of options:
+
+<img width="804" alt="gatsbot-example" src="https://user-images.githubusercontent.com/381801/52206647-c18ae980-2872-11e9-87ee-0905838206bd.png">
+
+Choosing an option would change the title or apply a label (or do something else) resulting in the PR being added to the changelog with the selected category applied.
+
+CI would be configured to fail until an option was chosen. Using a conventional commit title for your PR would result in a checkbox being preselected.
+
+Pros:
+
+- Allows contributors to categorise their own commits
+- Doesn't require any special knowledge (you don't need to read the docs to discover that you should format your PR title like X to have it included in the changelog)
+
+Cons:
+
+- This option will take the most amount of time to research, design and implement
+- Can we enforce this with CI?
+
+# Drawbacks
+
+The above options are mostly variations based on tradeoffs between time-to-implement and the quality of the end-result.
+
+# Alternatives
+
+Several alternatives are described above, including not making any changes.
+
+# Adoption strategy
+
+This RFC is mostly aimed at Gatsby maintainers and could be taught via the weekly maintainers meeting, automation / CI (where appropriate), and documentation on gatsbyjs.org.
+
+# How we teach this
+
+Similar to the adoption strategy. 
+
+Whatever option is chosen, it should be documented in the contributing section of gatsbyjs.org.
+
+# Unresolved questions
+
+There are a few unanswered questions listed in options 4 and 5. 
+
+### How to handle PR's with a single commit
+
+If a PR has only one commit, GitHub will use the commit message for the merge commit instead of the PR title. Will we need to work around this?
```

Comment:
If we enforce PR title, we can actually do a lot of stuff in changelog preset - i.e. if commit message itself is not following conventional commit, we can do GitHub API request to read additional metadata for commit, find PR and use PR title for changelog entry.

This would work for titles, but if we want to add additional information to changelog entries we would need to also figure out way to add additional notes in PR description and parse this information in changelog preset